container.go

```diff
@@ -157,6 +159,16 @@ func WithTeardown(t func() error) ContainerOption {
 	})
 }
 
+// WithLock allows for synchronizing access to the container execution so only
+// one is ever running at at time.  This is useful with OnError to prevent
+// multiple error handlers from running at the same time when parallel solves
+// are running.
+func WithLock(l *sync.Mutex) ContainerOption {
```

Comment:
Let's change this to take a `sync.Locker` so the caller has flexibility about what synchronization primitive to use, and can wrap the `Lock` and `Unlock` methods if desired.

Reply:
done